src/python/pants/engine/target.py

```diff
@@ -1612,12 +1616,26 @@ def can_generate(
     def path_globs(self, files_not_found_behavior: FilesNotFoundBehavior) -> PathGlobs:
         if not self.globs:
             return PathGlobs([])
-        error_behavior = files_not_found_behavior.to_glob_match_error_behavior()
+
+        # SingleSourceField has str as default type.
+        default_globs = (
+            [self.default] if self.default and isinstance(self.default, str) else self.default
+        )
```

Comment:
Was this a separate bug...?

Reply:
Ah, yes. Do you want the fix for that in a dedicated PR?

Reply:
I was going to say yes, if you're willing. Specifically so that we can cherry-pick to 2.8, where I introduced the bug. But I don't think this will actually impact anyone in reality? It's only if you explicitly set `source=Dockerfile` and the `Dockerfile` does not exist...on an experimental backend. Not worth the cherry-pick

So seems fine! But would be good to update PR description to mention the bug fix. (It's fine to not mention it in PR title imo)